.github/workflows/release.yaml

```diff
@@ -44,3 +61,68 @@ jobs:
           commit: 'chore(changesets): publish packages'
           publish: pnpm changeset:publish
           version: pnpm changeset:version
+
+      # Provide outputs for subsequent jobs
+      - name: Set outputs
+        run: |
+          echo "hasChangesets=${{steps.changesets.outputs.hasChangesets}}" >> $GITHUB_OUTPUTS
+          echo "published=${{steps.changesets.outputs.published}}" >> $GITHUB_OUTPUTS
+
+  # Update the versions.json after the release workflow
+  update-versions-on-release-pr:
+    runs-on: ubuntu-latest
+    needs: [release]
+    if: needs.release.outputs.main == true && needs.release.outputs.hasChangesets == true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: changeset-release/${{ github.ref_name }}
+
+      # Update versions.json
+      - name: Update versions.json
+        uses: ./.github/actions/update-versions
+
+      # On main: update versions.json, commit and push
+      - name: Push changes if on main
+        run: |
+          git add .
+          git commit -m "chore(release): update versions.json"
+          git push
+
+  open-versions-pr-against-main:
+    runs-on: ubuntu-latest
+    needs: [release]
+    if: needs.release.outputs.main == false && needs.release.outputs.hasChangesets == false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: changeset-release/${{ github.ref_name }}
+
+      # Update versions.json
+      - name: Update versions.json
+        uses: ./.github/actions/update-versions
```

Comment:
This action has the required inputs `version_file_path` and `package_file_path`.